tcl/org.testeditor.tcl.model/src/org/testeditor/tcl/util/TclModelUtil.xtend

```diff
@@ -321,11 +324,40 @@ class TclModelUtil extends TslModelUtil {
 		val templateContainer = testStep.templateContainer
 		val templateParameters = templateContainer?.template?.contents?.filter(TemplateVariable)
 		if (templateContainer !== null //
-			&& templateParameters !== null //
-			&& templateParameters.length > callParameterIndex) {
+		&& templateParameters !== null //
+		&& templateParameters.length > callParameterIndex) {
 			return templateParameters.drop(callParameterIndex).head
 		}
 		return null
 	}
-	
+
+	def dispatch boolean throwsFixtureException(Expression expression) {
+		// expressions may not throw FixtureExceptions, yet!
+		return false
+	}
+
+	def dispatch boolean throwsFixtureException(AssignmentThroughPath assignmentThroughPath) {
+		return assignmentThroughPath.expression.throwsFixtureException
+	}
+
+	def dispatch boolean throwsFixtureException(AssertionTestStep assertionTestStep) {
+		return assertionTestStep.assertExpression.throwsFixtureException
+	}
+
+	def dispatch boolean throwsFixtureException(TestStep testStep) {
+		val interaction = testStep.interaction
```

Comment:
… or you could use `getTemplateContainer(…)`, which returns either the interaction or the macro, and then use Xtend's switch with typeguards.

```
val container = testStep.templateContainer
switch(container) {
  InteractionType: return container.defaultMethod.operation.exceptions.exists[FixtureException.name.equals(qualifiedName)]
  Macro: return container.contexts.throwsFixtureException
  default: return false
}
```
Unrelated side note: isn't `findInteractionOrMacro` more or less the same as `getTemplateContainer`…?